src/adm/filter.rs

```diff
@@ -38,18 +38,28 @@ pub struct AdmFilter {
     pub filter_set: HashMap<String, AdmAdvertiserFilterSettings>,
 }
 
-/// Check that a given URL is valid according to it's corresponding filter
-fn check_url(url: Url, species: &'static str, filter: &[String]) -> HandlerResult<bool> {
-    let host = match url.host() {
-        Some(v) => v.to_string(),
-        None => {
-            return Err(HandlerErrorKind::MissingHost(species, url.to_string()).into());
+/// Extract the host from Url
+fn get_host(url: Url, species: &'static str) -> HandlerResult<String> {
+    url.host()
+        .map(|host| host.to_string())
+        .ok_or_else(|| HandlerErrorKind::MissingHost(species, url.to_string()).into())
```

Comment:
Out of curiosity, what's the etymology of the name `species` here?

Reply:
@jrconlin's knack for naming things?

Reply:
pretty much. Had to come up with the variant for what sort of host we're checking. "species" seemed reasonable, (as would "category", "classification", or some other version out of a thesaurus). I'm happy to accept any better suggestion.

Reply:
I have no better suggestions, just curious.